utils/gen_offsets.py

```diff
@@ -0,0 +1,89 @@
+"""
+Utility script to generate offset definitions for a peripheral from an SVD file.
+
+Usage:
+
+    python gen_offsets.py \
+        --svd <svd_file> \
+        --output <output_file> \
+        --peripheral <peripheral_name> \
+        --guard <header_guard> \
+        [--prefix <prefix>]
+
+Copyright (c) 2024 Nordic Semiconductor ASA
+SPDX-License-Identifier: Apache-2.0
+"""
+
+import argparse
+from datetime import datetime
+from pathlib import Path
+import sys
+import xml.etree.ElementTree as ET
+
+
+HEADER = """/*
+ * Copyright (c) {} Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* autogenerated using Nordic HAL utils/gen_offsets.py script */
+"""
+
+
+def parse_offsets(offsets, tree, base_name, base_offset):
+    if base_name:
+        base_name += "_"
+
+    entries = tree.findall("cluster") + tree.findall("register")
+    for entry in entries:
+        is_cluster = entry.tag == "cluster"
+        name = base_name + entry.find("name").text
+        offset = base_offset + int(entry.find("addressOffset").text, 16)
```

Comment:
Use base zero to parse both hex and decimal (all occurrences)
```suggestion
        offset = base_offset + int(entry.find("addressOffset").text, 0)
```

Reply:
does any address offset not use hex formatting?

Reply:
ok, thanks for the details, fixed